prow/tide/status.go

```diff
@@ -220,8 +223,21 @@ func requirementDiff(pr *PullRequest, q *config.TideQuery, cc contextChecker) (s
 // in order to generate a diff for the status description. We choose the query
 // for the repo that the PR is closest to meeting (as determined by the number
 // of unmet/violated requirements).
-func expectedStatus(queryMap *config.QueryMap, pr *PullRequest, pool map[string]PullRequest, cc contextChecker) (string, string) {
+func expectedStatus(queryMap *config.QueryMap, pr *PullRequest, pool map[string]PullRequest, cc contextChecker, blocks blockers.Blockers) (string, string) {
 	if _, ok := pool[prKey(pr)]; !ok {
+		// if the branch is blocked forget checking for a diff
```

Comment:
I was imagining that checking if the branch is blocked would be the last check in `requirementDiff` so that the status would still show PR authors info about how to get their PR closer to merging while the repo is temporarily blocked. I think this would be the preferred behavior for prow.k8s.io, but I can see why the current implementation would be preferred for blocker issues that are intended to last a long time or permanently.

I don't think it matters much now, but me might want to make this configurable in the future.

Reply:
FWIW, once (if) Prow starts using Checks API, the checks can be accompanied with a markdown text, so it would possible to expose all reasons why a PR is blocked.